File: Q1_LinkedListInsertion.c
#include <stdio.h>
#include <stdlib.h>
struct Node {int data; struct Node* next;};
void insert(struct Node** head, int val){struct Node* n=malloc(sizeof(struct Node));n->data=val;n->next=*head;*head=n;}
void print(struct Node* h){while(h){printf("%d ",h->data);h=h->next;}}
int main(){struct Node* head=NULL;insert(&head,3);insert(&head,2);insert(&head,1);print(head);}

File: Q1_LinkedListInsertion.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None
def insert(head,val):
    n=Node(val)
    n.next=head
    return n
def print_list(h):
    while h:
        print(h.data,end=" ")
        h=h.next
head=None
head=insert(head,3)
head=insert(head,2)
head=insert(head,1)
print_list(head)